tests/testpkg/debian/rules: Link with -Wl,--no-as-needed as the test
prog.c does not actually use any of the libadequate symbols.

diff -Nru adequate-0.9/tests/testpkg/debian/rules adequate-0.9ubuntu1/tests/testpkg/debian/rules
--- adequate-0.9/tests/testpkg/debian/rules	2013-12-04 09:52:00.000000000 +0100
+++ adequate-0.9ubuntu1/tests/testpkg/debian/rules	2013-12-18 08:28:42.000000000 +0100
@@ -17,14 +17,14 @@
 	ln -sf libadequate-lib.so.0 $(tmp)/libadequate-lib.so
 	$(CC) lib.c -fPIC -shared -Wl,-soname,libadequate-usrlib.so.0 -o $(tmp)/libadequate-usrlib.so.0
 	ln -sf libadequate-usrlib.so.0 $(tmp)/libadequate-usrlib.so
-	$(CC) prog.c -L$(tmp) -ladequate-lib -o $(tmp)/adequate-lib1
+	$(CC) prog.c -L$(tmp) -Wl,--no-as-needed -ladequate-lib -o $(tmp)/adequate-lib1
 	cd $(tmp) && seq 2 5 | xargs -t -I {} ln -f adequate-lib1 adequate-lib{}
-	$(CC) prog.c -L$(tmp) -ladequate-usrlib -o $(tmp)/adequate-usrlib1
+	$(CC) prog.c -L$(tmp) -Wl,--no-as-needed -ladequate-usrlib -o $(tmp)/adequate-usrlib1
 	cd $(tmp) && seq 2 5 | xargs -t -I {} ln -f adequate-usrlib1 adequate-usrlib{}
 	# incompatible-licenses
 	$(CC) lib.c -fPIC -shared -Wl,-soname,libssl.so.1.0.0 -o $(tmp)/libssl.so
 	$(CC) lib.c -fPIC -shared -Wl,-soname,libreadline.so.6 -o $(tmp)/libreadline.so
-	$(CC) prog.c -L$(tmp) -lssl -lreadline -o $(tmp)/adequate-license-conflict
+	$(CC) prog.c -L$(tmp) -Wl,--no-as-needed -lssl -lreadline -o $(tmp)/adequate-license-conflict
 	# missing-version-information
 	$(CC) -shared -Wl,--soname=lib$(at)-versionless.so.0 -Wl,--version-script=verscript-global lib.c -o $(tmp)/lib$(at)-versionless.so.0
 	ln -sf lib$(at)-versionless.so.0 $(tmp)/lib$(at)-versionless.so
